pkg/grpc/actions/auth/common.go

```diff
@@ -6,23 +6,50 @@ import (
 	"github.com/stretchr/testify/require"
 	"github.com/superplanehq/superplane/pkg/authorization"
 	"github.com/superplanehq/superplane/pkg/grpc/actions"
+	"github.com/superplanehq/superplane/pkg/models"
 	pbAuth "github.com/superplanehq/superplane/pkg/protos/authorization"
+	pbRoles "github.com/superplanehq/superplane/pkg/protos/roles"
+	pb "github.com/superplanehq/superplane/pkg/protos/users"
+	pbUsers "github.com/superplanehq/superplane/pkg/protos/users"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
-func convertRoleDefinitionToProto(roleDef *authorization.RoleDefinition, authService authorization.Authorization, domainID string) (*pbAuth.Role, error) {
+func convertRoleDefinitionToProto(roleDef *authorization.RoleDefinition, authService authorization.Authorization, domainID string, roleMetadataMap map[string]*models.RoleMetadata) (*pbRoles.Role, error) {
 	permissions := convertPermissionsToProto(roleDef.Permissions)
 
-	role := &pbAuth.Role{
-		Name:        roleDef.Name,
-		DomainType:  actions.DomainTypeToProto(roleDef.DomainType),
-		Permissions: permissions,
+	roleMetadata := roleMetadataMap[roleDef.Name]
+	role := &pbRoles.Role{
+		Metadata: &pbRoles.Role_Metadata{
+			Name:       roleDef.Name,
+			DomainType: actions.DomainTypeToProto(roleDef.DomainType),
+			DomainId:   domainID,
+			CreatedAt:  timestamppb.New(roleMetadataMap[roleDef.Name].CreatedAt),
+			UpdatedAt:  timestamppb.New(roleMetadataMap[roleDef.Name].UpdatedAt),
+		},
+		Spec: &pbRoles.Role_Spec{
+			DisplayName: models.GetRoleDisplayNameWithFallback(roleDef.Name, roleDef.DomainType, domainID, roleMetadata),
```

Comment:
Now we are ensuring that metadata is always available, we don't need this fallback mechanism, right?